ironic_prometheus_exporter/app/exporter.py

```diff
@@ -0,0 +1,30 @@
+import configparser
+import logging
+import os
+
+from flask import abort, Flask
+app = Flask(__name__)
+LOG = logging.getLogger(__name__)
+
+
+@app.route('/metrics', methods=['GET'])
+def prometheus_metrics():
+    try:
+        config = configparser.ConfigParser()
+        config.read(os.environ.get('IRONIC_CONFIG'))
+        DIR = config['oslo_messaging_notifications']['location']
+    except Exception as e:
+        LOG.error(e)
+        abort(404)
+
+    all_files = [os.path.join(DIR, name) for name in os.listdir(DIR)
+                 if os.path.isfile(os.path.join(DIR, name))]
+    content = ''
+    for file_name in all_files:
+        with open(file_name, 'r') as file:
+            content += ''.join(file.readlines())
```

Comment:
Can we stream content as we generate it? Given that it's a string, it should not be hard.

Reply:
Going to look how to do this, but would be necessary? I think when Prometheus request he wan'ts to get all the content at once...

Reply:
Just to avoid building a potentially large string in memory.

Reply:
You could also use file.read() to skip splitting it into lines, since you're just joining them back together again anyway.

Reply:
I've switched to use the streaming from Flask  http://flask.pocoo.org/docs/1.0/patterns/streaming/